### PR TITLE
Create documented_ZPrime_prediction.py

### PR DESCRIPTION
**Pull request by ardabtank**
There are several changes and updates in these code:

- Documented in detail.

- RandomUnderSampler is used to for balancing the datasets. 

- Added a "checking.txt" file to save some important checks.

- StratifiedKFold is used instead of "train_test_split" for splitting the dataset into three parts. Validation data is now "X_val" and "X_test" is used for further predictions.

- Added a function called "check_intersections" which checks if there are any instersection between three splitted parts.

- The stabilized neural network is changed with a hyperparameter optimization class. 

- Leaky ReLU is considered as an activation function.

- The explanations of metrics removed.

### CODE REVIEW

**Review by ardabtank — COMMENTED** [review]
Forgotten update:

- Metric graphs are now saving into "Graphs" folder as png.
